eclipse-theia/theia
# Add Memory Inspector to Theia Repo

#### What it does


This PR brings the code for inspecting memory on the frontend into the main Theia repo from the [Theia CPP Debug](https://github.com/eclipse-theia/theia-cpp-extensions) repository. Previously, that repository included a number of CPP-specific language features, but changes in Theia and in that repository have made it language-agnostic for the most part. At the same time, keeping it in a separate repo generates [its own problems](https://github.com/eclipse-theia/theia/issues/10042), so that deprecating that repository and moving the code here seems the best option.

#### How to test


1. Install the CDT-GDB VSCode plugin.
2. You should be able to run through the set of steps outlined [here](https://github.com/eclipse-theia/theia-cpp-extensions/pull/119), substituting `packages/memory-inspector/memory-inspector-example-workspace` for the `cpp-debug-workspace` of that repository.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: Colin Grant <colin.grant@ericsson.com>
